articles/browser-extension-basics.md

```diff
@@ -0,0 +1,29 @@
+## Browser Extension Basics
+
+Useful Resources:
+
+- [MDN Anatomy of an Extension](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension)
+- [Mozilla Extension Workshop](https://extensionworkshop.com/)
+- [Chrome Developer Overview](https://developer.chrome.com/extensions/overview)
+- [Chrome Developer Guide](https://developer.chrome.com/extensions/devguide)
+
+
+### Standard Extension Architecture
+
+1. `manifest.json` ([Firefox](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json), [Chrome](https://developer.chrome.com/extensions/manifest)) is the only required file for an extension. The main purposes of the manifest are as follows:
+    - specify extension metadata, e.g. `name`, `version`, `locale` etc.
+    - point to core files, e.g. `background`, `content_scripts`, `devtools_page`, `options` and bundled `web_accessible_resources`
+    - assert permissions on browser APIs and other resources, urls etc. e.g.`storage`, `tabs`, `bookmarks`, `https://*/*`
+    - declare UI elements e.g. `browser_action` icon
+
+2. Content ([Firefox](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts), [Chrome](https://developer.chrome.com/extensions/content_scripts)): The content scripts (`.js` or `.css`) are loaded into and executed within the context of a given web page. They have access to the DOM and as such can read and manipulate the existing HTML. These scripts can dispatch XHRs. In order to interact with most other browser APIs/resources or with other extension elements (e.g. to tell them about certain DOM elements present) they must usually communicate via the intermediary of a `background script`, through message passing (see below). In order to communicate with a page `<script>` (or vice-versa) the `window.postMessage()` API must be used.
```

Comment:
Also maybe mention that you need to add permissions to the manifest to be able to inject a content script.

Reply:
👍